test/e2eexternal/dataplane/connectivity.go

```diff
@@ -0,0 +1,335 @@
+		if framework.TestContext.ClusterIDs[idx] != "cluster1" {
+			Expect(podLog).To(ContainSubstring(dockerIP))
+		}
+		// temporary debug log
+		framework.Logf("podLog: %v", podLog)
+
+		By(fmt.Sprintf("Sending http request from test pod %q %q in cluster %q to exteranl app %q",
+			pod.Name, podIP, framework.TestContext.ClusterIDs[idx], dockerIP))
+
+		config := framework.RestConfigs[idx]
+
+		err = sendHTTPRequestFromPod(cs, config, f.Namespace, pod.Name, testContainerName, dockerIP, 80)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verifying that docker get request")
+
+		dockerLog, err := getLogFromDocker(extAppName)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		// TODO: For cluster1 (cluster that is connected to external network),
+		// source IP seems to become the node IP of external network CIDRs, like 172.16.0.3
+		// not the dockerIP (external app's IP). This needs to be investigated further, before merge.
+		if framework.TestContext.ClusterIDs[idx] != "cluster1" {
+			Expect(dockerLog).To(ContainSubstring(podIP))
+		}
+		// temporary debug log
+		framework.Logf("dockerLog: %v", dockerLog)
+	}
+}
+
+func makePodSpec(nameBase string, label map[string]string, containerName, image string, command []string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: nameBase,
+			Labels:       label,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    containerName,
+					Image:   image,
+					Command: command,
+				},
+			},
+		},
+	}
+}
+
+func createPod(cs *kubeclientset.Clientset, namespace string, podSpec *v1.Pod) (*v1.Pod, error) {
+	pod, err := cs.CoreV1().Pods(namespace).Create(podSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	return pod, nil
+}
+
+func waitForPodReady(cs *kubeclientset.Clientset, namespace, name string) error {
+	return wait.PollImmediate(
+		5*time.Second,  /* interval */
+		60*time.Second, /* timeout */
+		func() (bool, error) {
+			pod, err := cs.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			switch pod.Status.Phase {
+			case v1.PodRunning:
+				for _, status := range pod.Status.ContainerStatuses {
+					if !status.Ready {
+						return false, nil
+					}
+				}
+				return true, nil
+			case v1.PodFailed, v1.PodSucceeded:
+				return false, fmt.Errorf("pod won't become running (current phase: %v)", pod.Status.Phase)
+			}
+
+			return false, nil
+		})
+}
+
+func makeServiceSpec(svcName string, label map[string]string, port, targetPort int) *v1.Service {
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: svcName,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{{
+				Port:       int32(port),
+				TargetPort: intstr.FromInt(targetPort),
+				Protocol:   v1.ProtocolTCP,
+			}},
+			Selector: label,
+		},
+	}
+}
+
+func createService(cs *kubeclientset.Clientset, namespace string, svcSpec *v1.Service) (*v1.Service, error) {
```

Comment:
https://github.com/submariner-io/shipyard/blob/devel/test/e2e/framework/services.go

You're replicating part of the framework functionality here ^

Let's use the functions already available on the framework.